lib/ortho_functions.py

```diff
@@ -1816,6 +1826,78 @@ def overlap_check(geometry_wkt, spatial_ref, demPath):
 
     return overlap
 
+def check_image_overlap_with_gpkg(geometry_wkt, spatial_ref, gpkg_path):
+    """
+    Parameters:
+        image_geometry (wkt)
+        image_spatial_ref
+        gpkg_path (str): Path to the GeoPackage file.
+
+    Returns:
+        str: The value of the "dempath" field from the final overlapping layer, or None if no overlap is found.
+    """
+    # image properties
+    imageSpatialReference = spatial_ref.srs
+    image_geometry = ogr.CreateGeometryFromWkt(geometry_wkt)
+
+    # Open the GeoPackage dataset
+    dataset = gdal.OpenEx(gpkg_path, gdal.OF_VECTOR)
```

Comment:
Check if this way of opening the dataset can handle concurrency well.

Reply:
@SAliHossaini , Did you check if the gdal.OpenEx method handles concurrency (two people opening the dataset at the same time)?

Reply:
Yes, it should be good for read-only operations without issues. I and Jesse tested concurrency and it worked for this function.